### PR TITLE
Coupling measurement improvements

### PR DESCRIPTION
**Pull request by murilobalves**
- Rename parameters from apsuite's `MeasCoupling` class that is used in the corresponding GUI, following changes from this PR: https://github.com/lnls-fac/apsuite/pull/210.
- Rearrange options in Measurement Control in two columns
- Create Apply Achromatic ΔKsL part to adjust betatron coupling
- Add guiding lines with expected tunes related to lower and upper limit of current variations of quadrupole. The expected variations were obtained with the following code: 

```
from pymodels import si
import pyaccel
import numpy as np

mod = si.create_accelerator()
fam = si.get_family_data(mod)
quadname = si.families.families_quadrupoles()
twiss, *_ = pyaccel.optics.calc_twiss(mod)
dnus = dict()

for nam in quadname:
   idx = fam[nam]['index']
   nquads = len(idx)
   kl = mod[idx[0][0]].KL
   betax = twiss.betax[idx[0][0]]
   betay = twiss.betay[idx[0][0]]
   rel_dnux = betax*nquads*kl/4/np.pi
   rel_dnuy = -betay*nquads*kl/4/np.pi
   dnus[nam] = (rel_dnux, rel_dnuy)

print(dnus)
```
The dictionary `dnus` contains the relative tunes variations for each quadrupole family. The value corresponds to 

image

This obtained considering the variation in quad strength is dKL = p (KL)_0, where p is the relative variation p = dKL/(KL)_0 and it is assumed that it is 1:1 to the relative variation in the quad family current, i.e., p = dI/I_0 as well. 

To obtain the absolute tune variation, one must multiply each value by the quadrupole family relative strength variation. For example, one dict entry is `'Q3': (11.52, -13.38)`. This means that if the strength of Q3 family is increased by 0.1%, the expected change of tunex is `11.52 * 0.001 = 0.01152` and of tuney is `-13.38 * 0.001 = - 0.01338`.

New window:

image

### CODE REVIEW

**Review by fernandohds564 — CHANGES_REQUESTED** [review]

**Comment by xresende** [comment]
@murilobalves , although the relationship between coupling and correction factor is not exactly known, or well defined in general, we already have a good correlation estimate for the range we usually apply, right? should we not use 'coupling' itself in the interface? otherwise the user will have to remember or search elsewhere what factor he/she should use to approach the coupling in mind. Besides a LineEdit with coupling I would add a Label with the factor so that new users of this functionality can gain a feeling for the correlation along time.

**Comment by fernandohds564** [comment]
> @murilobalves , although the relationship between coupling and correction factor is not exactly known, or well defined in general, we already have a good correlation estimate for the range we usually apply, right? should we not use 'coupling' itself in the interface? otherwise the user will have to remember or search elsewhere what factor he/she should use to approach the coupling in mind. Besides a LineEdit with coupling I would add a Label with the factor so that new users of this functionality can gain along time a feeling for the correlation.

@xresende, there is a tooltip in the 'Factor' label that says: 'The relation between Factor and ΔCoupling is roughly 1:1'.
Do you think this is enough?

@murilobalves started with these labels indicating coupling instead of KsL. We changed it because we can apply negative values of the factor, which does not map into coupling directly.
The option o apply negative values came from the possible need of undoing some previously applied strength.

**Review by fernandohds564 — APPROVED** [review]

**Review by anacso17 — APPROVED** [review]

**Comment by xresende** [comment]
another suggestion: in the 'save and load' panel, is it not better to allign both buttons horizontally? this should save vertical space and the plot area may be increased a bit. are you following a standard used in other windows ?

**Comment by murilobalves** [comment]
> another suggestion: in the 'save and load' panel, is it not better to allign both buttons horizontally? this should save vertical space and the plot area may be increased a bit. are you following a standard used in other windows ?

this space is reserved to the filename string, when some data is loaded or saved, this is why there is this seemly empty area

**Comment by fernandohds564** [comment]
> > > @murilobalves , although the relationship between coupling and correction factor is not exactly known, or well defined in general, we already have a good correlation estimate for the range we usually apply, right? should we not use 'coupling' itself in the interface? otherwise the user will have to remember or search elsewhere what factor he/she should use to approach the coupling in mind. Besides a LineEdit with coupling I would add a Label with the factor so that new users of this functionality can gain along time a feeling for the correlation.
> > 
> > 
> > @xresende, there is a tooltip in the 'Factor' label that says: 'The relation between Factor and ΔCoupling is roughly 1:1'. Do you think this is enough? @xresende we added a message to leave it even more explicit. Is it ok now?
> > @murilobalves started with these labels indicating coupling instead of KsL. We changed it because we can apply negative values of the factor, which does not map into coupling directly. The option o apply negative values came from the possible need of undoing some previously applied strength.
> 
> I think it is ok, @fernandohds564, @murilobalves !
> 
> but another thing that came to my mind: when we think of coupling we think in percentage, usually. I think the factor in the interface is _not_ in percentage, right? it it is the case, the specific string '1:1 to Coupling' does not address this... but I do not have any suggestion on how to improve it without being verbose...

@xresende  we changed the string so the units become more explicit. What do you think?

**Comment by xresende** [comment]
@murilobalves , @fernandohds564 , adding a tune|q3 response matrix from model to plot expected tune variations (for zero or nominal 1% coupling) is too complicated? :grimacing: i think it could help a lot users to decide on Q3 variation parameters before messing with the machine magnetic history...

**Comment by murilobalves** [comment]
> @murilobalves , @fernandohds564 , adding a tune|q3 response matrix from model to plot expected tune variations (for zero or nominal 1% coupling) is too complicated? grimacing i think it could help a lot users to decide on Q3 variation parameters before messing with the machine magnetic history...

@fernandohds564 and I discussed and we think it is a good idea indeed and not too complicated to implement, I'll work on that!

File: siriushlafac/VERSION [hunk 1]
```diff
@@ -1 +1 @@
-0.8.2
+0.8.3
```
Comment (anacso17) [inline siriushlafac/VERSION:1]: It seems the changes here bring a `0.9.0`

Comment (murilobalves) [reply siriushlafac/VERSION]: done!

**Pull request merged**